lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -5210,6 +5224,14 @@ struct ExportSplitVerilogPass
     // on the command line.
     // TODO: This should be moved up to circt-opt and circt-translate.
     applyLoweringCLOptions(getOperation());
+
+    // Prepare the ops in the module for emission.
+    mlir::OpPassManager preparePM("builtin.module");
+    auto &modulePM = preparePM.nest<hw::HWModuleOp>();
+    modulePM.addPass(createPrepareForEmissionPass());
+    if (failed(runPipeline(preparePM, getOperation())))
+      signalPassFailure();
+
     if (failed(exportSplitVerilog(getOperation(), directoryName)))
```

Comment:
```suggestion
    if (failed(exportSplitVerilogImpl(getOperation(), directoryName)))
```
So the prepare isn't run twice.